src/lib/core/evio.c

```diff
@@ -404,19 +404,35 @@ evio_service_bind(struct evio_service *service, const char *uri)
 /** It's safe to stop a service which is not started yet. */
 void
 evio_service_stop(struct evio_service *service)
+{
+	evio_service_deactivate(service);
+	if (service->ev.fd >= 0) {
+		int fd = service->ev.fd;
+		if (close(service->ev.fd) < 0) {
+			diag_set(SocketError, sio_socketname(fd), "close");
+			diag_log();
+			diag_clear(diag_get());
+		}
+		ev_io_set(&service->ev, -1, 0);
+		if (service->addr.sa_family == AF_UNIX) {
```

Comment:
It's very difficult. In fact both close and delete unix socket path happens in file iproto.cc:2574
/*
 * Here we close socket and unlink unix socket path.
 * in case it's unix socket.
 */
if (evio_service_is_active(&binary))
	evio_service_stop(&binary);

Here i only log error, to check it in the test

Reply:
I don't think it is ok. Assume, there is another unix server running - then you will delete its socket file. Does not look right. Deleting the socket in the end of usage seems better.